### PR TITLE
Make DestroyDir destroy directories recursively

### PR DESCRIPTION
**Pull request by cheng-chang**
Currently, `DeleteDir` only deletes the directory if there are no other directories under the target dir. This PR makes it delete directories recursively.

Test Plan:
Added a new unit test in testutil_test.cc.
`make testutil_test`

### CODE REVIEW

**Comment by zhichao-cao** [comment]
Thanks for the changes! Some comments left.

File: CMakeLists.txt [hunk 1]
```diff
@@ -1090,6 +1090,7 @@ if(WITH_TESTS)
         table/sst_file_reader_test.cc
         table/table_test.cc
         table/block_fetcher_test.cc
+        test_util/testutil_test.cc
```
Comment (zhichao-cao) [inline CMakeLists.txt:1093]: run make format

File: TARGETS [hunk 2]
```diff
@@ -1485,6 +1485,13 @@ ROCKS_TESTS = [
         [],
         [],
     ],
+    [
```
Comment (zhichao-cao) [inline TARGETS:1488]: Did you run python buckifier/buckify_rocksdb.py or manually changed it?

Comment (cheng-chang) [reply TARGETS:1488]: buckify

**Pull request closed without merging**